olive/engine/packaging/packaging_generator.py

```diff
@@ -11,51 +11,65 @@
 from collections import OrderedDict
 from pathlib import Path
 from string import Template
+from typing import Dict
 
 import pkg_resources
 
 from olive.common.utils import run_subprocess
 from olive.engine.footprint import Footprint
 from olive.engine.packaging.packaging_config import PackagingConfig, PackagingType
+from olive.hardware import AcceleratorSpec
 
 logger = logging.getLogger(__name__)
 
 
 def generate_output_artifacts(
-    packaging_config: PackagingConfig, foot_print: Footprint, pf_footprint: Footprint, output_dir: Path
+    packaging_config: PackagingConfig,
+    footprints: Dict[AcceleratorSpec, Footprint],
+    pf_footprints: Dict[AcceleratorSpec, Footprint],
+    output_dir: Path,
 ):
-    if pf_footprint.nodes is None or len(pf_footprint.nodes) == 0:
+    if (
+        sum([len(f.nodes) for f in footprints.values()]) == 0
```

Comment:
Nit: we only need to verify pf_footprints node length as pf_footprint is a subset of footprint